test/e2e/info_api_test.go

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package e2e_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/cortexproject/cortex/integration/e2e"
+	e2edb "github.com/cortexproject/cortex/integration/e2e/db"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/thanos-io/thanos/pkg/objstore/client"
+	"github.com/thanos-io/thanos/pkg/objstore/s3"
+	"github.com/thanos-io/thanos/pkg/query"
+	"github.com/thanos-io/thanos/pkg/runutil"
+	"github.com/thanos-io/thanos/pkg/testutil"
+	"github.com/thanos-io/thanos/test/e2e/e2ethanos"
+)
+
+func TestInfo(t *testing.T) {
+	t.Parallel()
+
+	s, err := e2e.NewScenario("e2e_test_info")
+	testutil.Ok(t, err)
+	t.Cleanup(e2ethanos.CleanScenario(t, s))
+
+	prom1, sidecar1, err := e2ethanos.NewPrometheusWithSidecar(s.SharedDir(), "e2e_test_info", "alone1", defaultPromConfig("prom-alone1", 0, "", ""), e2ethanos.DefaultPrometheusImage())
+	testutil.Ok(t, err)
+	prom2, sidecar2, err := e2ethanos.NewPrometheusWithSidecar(s.SharedDir(), "e2e_test_info", "alone2", defaultPromConfig("prom-alone2", 0, "", ""), e2ethanos.DefaultPrometheusImage())
+	testutil.Ok(t, err)
+	prom3, sidecar3, err := e2ethanos.NewPrometheusWithSidecar(s.SharedDir(), "e2e_test_info", "alone3", defaultPromConfig("prom-alone3", 0, "", ""), e2ethanos.DefaultPrometheusImage())
+	testutil.Ok(t, err)
+	testutil.Ok(t, s.StartAndWaitReady(prom1, sidecar1, prom2, sidecar2, prom3, sidecar3))
+
+	m := e2edb.NewMinio(8080, "thanos")
+	testutil.Ok(t, s.StartAndWaitReady(m))
+	str, err := e2ethanos.NewStoreGW(s.SharedDir(), "1", client.BucketConfig{
+		Type: client.S3,
+		Config: s3.Config{
+			Bucket:    "thanos",
+			AccessKey: e2edb.MinioAccessKey,
+			SecretKey: e2edb.MinioSecretKey,
+			Endpoint:  m.NetworkHTTPEndpoint(),
+			Insecure:  true,
+		},
+	})
+	testutil.Ok(t, err)
+	testutil.Ok(t, s.StartAndWaitReady(str))
+
+	// Register 1 sidecar using `--store`.
+	// Register 2 sidecars and 1 storeGW using `--endpoint`.
+	// Register `sidecar3` twice to verify it is deduplicated.
+	q, err := e2ethanos.NewQuerierBuilder(s.SharedDir(), "1", []string{sidecar1.GRPCNetworkEndpoint()}).
```

Comment:
If `endpoint` and specific stores are configured (`metadata`, `target`, etc) together, does it work? Can you add this test case?

Reply:
Yes sure. I will add this test case.